Particle expiry condition not working

### Issue description

I have a particle that I want to kill when it touches any block (other than air). The only way I could think of doing this is to turn off collision and have it check if it's in a block. The problem is that I get a gradle error in my particle class, and when I inspected the code, it passes on a nonexistent world argument to the procedure.
This is the bad code:
if (LaserParticleExpireProcedure.execute(world, x, y, z, onGround))
			this.remove();
The "world" variable it's using isn't declared anywhere. It just doesn't exist except for this one instance.
Note: In the log, I added a few more spaces so the arrow under the code would align like it does for me.

### How to reproduce this issue?

Create a particle.
Add an expiry procedure.
Have it check if the block it's at is solid. If it is, return true, otherwise return false.
When saved, it should cause a gradle error in the particle class.

### Operating system

Windows

### Details

MCreator version 2023.2
MC version is 1.19.2

### Example workspace

_No response_

### Logs

Executing Gradle task: build 
Build info: MCreator 2023.2.24119, forge-1.19.2, 64-bit, 16015 MB, Windows 11, JVM 17.0.5, JAVA_HOME: C:\Program Files\Pylo\MCreator\jdk, started on: 2023-08-08-11:10:08 
  
> Configure project : 
The code of this workspace uses official obfuscation mappings provided by Mojang. These mappings fall under their associated license you should be fully aware of. 
(c) 2020 Microsoft Corporation. These mappings are provided "as-is" and you bear the risk of using them. You may copy and use the mappings for development purposes, 
but you may not redistribute the mappings complete and unmodified. Microsoft makes no warranties, express or implied, with respect to the mappings provided here. 
Use and modification of this document or the source code (in any form) of Minecraft: Java Edition is governed by the Minecraft End User License Agreement available 
at https://account.mojang.com/documents/minecraft_eula. 
  
> Task :compileJava FAILED 
[C:\Users\redst\MCreatorWorkspaces\entropy\src\main\java\net\entropy\client\particle\LaserParticleParticle.java:78](file:///C:/Users/redst/MCreatorWorkspaces/entropy/src/main/java/net/entropy/client/particle/LaserParticleParticle.java:78): error: cannot find symbol 
      if (LaserParticleExpireProcedure.execute(world, x, y, z, onGround)) 
                                                                      ^ 
  symbol:   variable world 
  location: class LaserParticleParticle 
1 error 
FAILURE: Build failed with an exception. 
* What went wrong: 
Execution failed for task ':compileJava'. 
> Compilation failed; see the compiler error output for details. 
* Try: 
> Run with --stacktrace option to get the stack trace. 
> Run with --info or --debug option to get more log output. 
> Run with --scan to get full insights. 
* Get more help at https://help.gradle.org 
BUILD FAILED in 2s 
1 actionable task: 1 executed 
  
BUILD FAILED 
Task completed in 4 seconds

### Issue tracker rule checks (please read carefully)

- [X] I have checked that my problem <a href='https://github.com/MCreator/MCreator/issues?q=is%3Aissue' target='_blank'>is not already reported</a>
- [X] I have checked that my problem is not covered on <a href='https://mcreator.net/support/knowledgebase' target='_blank'>Knowledge Base</a> or on <a href='https://mcreator.net/wiki' target='_blank'>MCreator's Wiki</a>
- [X] I have checked that my written content does not violate the <a href='https://mcreator.net/wiki/general-publishing-guidelines' target='_blank'>publishing guidelines</a>

--- a/plugins/generator-1.19.2/forge-1.19.2/templates/particle.java.ftl
+++ b/plugins/generator-1.19.2/forge-1.19.2/templates/particle.java.ftl
@@ -118,9 +118,6 @@ package ${package}.client.particle;
 
 		<#if hasProcedure(data.additionalExpiryCondition)>
 		Level world = this.level;
-		double x = this.x;
-		double y = this.y;
-		double z = this.z;
 		if (<@procedureOBJToConditionCode data.additionalExpiryCondition/>)
 			this.remove();
 		</#if>

--- a/plugins/generator-1.19.2/forge-1.19.2/templates/particle.java.ftl
+++ b/plugins/generator-1.19.2/forge-1.19.2/templates/particle.java.ftl
@@ -117,6 +117,7 @@ package ${package}.client.particle;
 		</#if>
 
 		<#if hasProcedure(data.additionalExpiryCondition)>
+		Level world = this.level;
 		double x = this.x;
 		double y = this.y;
 		double z = this.z;

--- a/plugins/generator-1.19.4/forge-1.19.4/templates/particle.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/templates/particle.java.ftl
@@ -118,6 +118,7 @@ package ${package}.client.particle;
 		</#if>
 
 		<#if hasProcedure(data.additionalExpiryCondition)>
+		Level world = this.level;
 		double x = this.x;
 		double y = this.y;
 		double z = this.z;

--- a/plugins/generator-1.19.4/forge-1.19.4/templates/particle.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/templates/particle.java.ftl
@@ -119,9 +119,6 @@ package ${package}.client.particle;
 
 		<#if hasProcedure(data.additionalExpiryCondition)>
 		Level world = this.level;
-		double x = this.x;
-		double y = this.y;
-		double z = this.z;
 		if (<@procedureOBJToConditionCode data.additionalExpiryCondition/>)
 			this.remove();
 		</#if>

--- a/plugins/generator-1.20.1/forge-1.20.1/templates/particle.java.ftl
+++ b/plugins/generator-1.20.1/forge-1.20.1/templates/particle.java.ftl
@@ -118,6 +118,7 @@ package ${package}.client.particle;
 		</#if>
 
 		<#if hasProcedure(data.additionalExpiryCondition)>
+		Level world = this.level;
 		double x = this.x;
 		double y = this.y;
 		double z = this.z;

--- a/plugins/generator-1.20.1/forge-1.20.1/templates/particle.java.ftl
+++ b/plugins/generator-1.20.1/forge-1.20.1/templates/particle.java.ftl
@@ -119,9 +119,6 @@ package ${package}.client.particle;
 
 		<#if hasProcedure(data.additionalExpiryCondition)>
 		Level world = this.level;
-		double x = this.x;
-		double y = this.y;
-		double z = this.z;
 		if (<@procedureOBJToConditionCode data.additionalExpiryCondition/>)
 			this.remove();
 		</#if>

